File: javascript/comerciante.js
const n = parseInt(prompt("Serao digitados dados de quantos produtos?"));

let nomes = [], valoresCompra = [], valoresVenda = [], lucros = [];

let abaixo10 = 0, entre20e30 = 0, acima20 = 0, totalCompra = 0, totalVenda = 0, totalLucro;

for(i = 0; i < n; i++) {
    alert("Produto " + (i + 1));

    nomes[i] = prompt("Nome:")
    valoresCompra[i] = parseFloat(prompt("Preco de compra:"));
    valoresVenda[i] = parseFloat(prompt("Preco de venda:"));

    lucros[i] = (valoresVenda[i] - valoresCompra[i]) / valoresCompra[i];
}

for (i = 0; i < n; i++) {
    if (lucros [i] < 0.1) {
        abaixo10++;
    } else if (lucros[i] < 0.2) {
        entre20e30++;
    } else {
        acima20++;
    }
}

for (i = 0; i < n; i++) {
    totalCompra += valoresCompra[i];
    totalVenda += valoresVenda[i];
}

totalLucro = totalVenda - totalCompra;

console.log("RELATORIO: ");
console.log("Lucro abaixo de 10%%: %d", abaixo10);
console.log("Lucro entre 10%% e 20%%: %d", entre20e30);
console.log("Lucro acima de 20%%: %d", acima20);
console.log("Valor total de compra: %.2f", totalCompra);
console.log("Valor total de venda: %.2f", totalVenda);
console.log("Lucro total: %.2f", totalLucro);

File: python/comerciante.py
n = int(input("Serao digitados dados de quantos produtos? "))

nomes = []
precos_compra = []
precos_venda = []

for i in range(0, n):
    print(f"Produto {i+1}:")
    nomes.append(input("Nome: "))
    precos_compra.append(float(input("Preco de compra: ")))
    precos_venda.append(float(input("Preco de venda: ")))

valor_total_compra = 0
valor_total_venda = 0
lucro_abaixo_10 = 0
lucro_entre_10e20 = 0
lucro_acima_20 = 0
for i in range(0, n):
    lucro = (precos_venda[i] - precos_compra[i]) / precos_compra[i]

    if lucro < 0.1:
        lucro_abaixo_10 += 1
    elif lucro <= 0.2:
        lucro_entre_10e20 += 1
    else:
        lucro_acima_20 += 1

    valor_total_venda += precos_venda[i]
    valor_total_compra += precos_compra[i]

lucro_total = valor_total_venda - valor_total_compra

print("RELATORIO:")
print("Lucro abaixo de 10%:", lucro_abaixo_10)
print("Lucro entre 10% e 20%:", lucro_entre_10e20)
print("Lucro acima de 20%:", lucro_acima_20)
print(f"Valor total de compra: {valor_total_compra:.2f}")
print(f"Valor total de venda: {valor_total_venda:.2f}")
print(f"Lucro total: {lucro_total:.2f}")
